### ISSUE TITLE
How to make an existing origin private

### ISSUE THREAD

**Issue by bdangit** [habitat-sh/builder#1008]
Does Builder support the ability to switch origins to private visibility after it has been created as public visibility?

If an origin goes private, does that mean any user/client has to be part of that origin to gain at least read access to the packages?

**Comment by raskchanky** [comment]
Origins themselves aren't public or private.  Origins have a default visibility setting that controls what visibility is assigned to new packages that are built.  That setting can be changed via the HTTP API like so:

```
curl -v \
-X PUT \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $HAB_AUTH_TOKEN" \
-d '{"default_package_visibility":"private"}' \
https://bldr.habitat.sh/v1/depot/origins/foo
```

This example would change the default package visibility for the `foo` origin to `private` from whatever it currently is (presumably `public`).  Changing this setting has no effect on existing packages, only on new packages that are built after the change was made.

Once the default package visibility setting is changed to private, new package builds will be private, and users will need to be members of the origin to see those builds.  Existing package builds that were done before the change remain visible to everyone.

**Comment by dmccown** [comment]
Thanks for the feedback @bdangit. Does the comment above help unblock you? We are looking at options for having better CLI and UI support to solve your use case. 

If you are unblocked we'll close this issue and link it with issues that can make this user experience better.

**Comment by bdangit** [comment]
Ah this will help me and it won’t help me at the same time. 😅 

I have compounded issues.  While better UI support will help, I’m now put in a position where I have to add members to that origin and that is something I wanted to avoid until RBAC is supported. 

I trust my users but I don’t;  we need mechanisms put in place to prevent goof ups, which is another story.

In the interim, I’ll accept @raskchanky’s answer.  Thanks!
